.travis.yml

```diff
@@ -0,0 +1,57 @@
+language: c
+dist: trusty   # Travis build machine is Ubuntu Trusty 14.04 x64
+sudo: required # A sudo enabled, full VM is used for building instead of a docker container
+
+git:
+  submodules: false
+
+before_install:
+  - sudo apt-get update -qq
+  # Pre-reqs for building dsc
+  - sudo apt-get install -qq pkg-config make bison g++ rpm librpm-dev libpam0g-dev libssl-dev libmysqlclient-dev 
+
+# Travis-CI automatically clones the Powershell-DSC-for-Linux repo since this is where this .travis.yml file is located
+# We will ignore this clone and clone the superproject Build-Powershell-DSC-for-Linux as follows
+
+install:
+  - cd ..
+  - git clone https://github.com/Microsoft/Build-Powershell-DSC-for-Linux.git ./Build-Powershell-DSC-for-Linux
+  - cd ./Build-Powershell-DSC-for-Linux
+  # change url from ssh to https for submodules, travis-ci cannot download a submodule that is on a public url.
+  - sed -i 's/git@github.com:/https:\/\/github.com\//' .gitmodules 
+  - git submodule update --init --recursive dsc omi pal
+  - # Do the same for ostc-openssl
+  - cd ..
+  - git clone https://github.com/Microsoft/ostc-openssl.git ./ostc-openssl
+  - cd ./ostc-openssl
+  - sed -i 's/git@github.com:/https:\/\/github.com\//' .gitmodules 
+  - git submodule update --init --recursive openssl-0.9.8 openssl-1.0.0
+  - cd ../Build-Powershell-DSC-for-Linux
```

Comment:
Holy smokes, that's a lot of crud (and complexity) to deal with the superproject. In SCXcore, I have:

```
  - # Fetch the Build-SCXcore repository for dependencies to build from
  - git config --global 'url.https://github.com/.insteadOf' 'git@github.com:'
  - git clone --recursive 'https://github.com/Microsoft/Build-SCXcore.git' ../bld-scxcore
  -
  - # Master branch for all submodules
  - cd ../bld-scxcore
  - git submodule foreach git checkout master
  -
  - # Patch in the version of SCXcore that Travis gave to us
  - rm -rf opsmgr
  - cp -R ../SCXcore opsmgr
```

I think that's a lot more simple. For example, why use `sed` to modify .gitmodules when you can just use a `git` command to favor https?

Also, building SSL is obviously doable, but can't you build DSC to just use the local SSL on the system? That would be a lot faster and easier. We build OpenSSL once per build system and install it. It seems foolish to build it for every CI run when you can just abstract that out and use the SSL that is already installed.